dags/nyc_taxi_dag.py

```diff
@@ -1,138 +1,65 @@
-
-    # flatten nested list as there is only one column
-    return list(map(lambda record: record[0], records))
-
-
-def clean_data_urls(ti):
-    data_urls_raw = ti.xcom_pull(task_ids="download_data_urls", key="return_value")
-
-    data_urls = data_urls_raw.split("\n")
-    # we only import Yellow tripdata for now due to different CSV schemas
-    data_urls_filtered = filter(lambda element: "yellow" in element, data_urls)
-
-    return list(data_urls_filtered)
-
-
-def identitfy_missing_urls(ti):
-    data_urls_processed = ti.xcom_pull(task_ids="get_processed_files")
-    data_urls_available = ti.xcom_pull(task_ids="clean_data_urls")
-
-    return list(set(data_urls_available) - set(data_urls_processed))
-
-
-def process_new_files(ti):
-    missing_urls = ti.xcom_pull(task_ids="identitfy_missing_urls")
-
-    for missing_url in missing_urls:
-        logging.info(missing_url)
-
-        file_name = missing_url.split("/").pop()
-
-        PostgresOperator(
-            task_id=f"copy_{file_name}",
-            postgres_conn_id="cratedb_demo_connection",
-            sql=f"""
-                    COPY nyc_taxi.load_trips_staging
-                    FROM '{missing_url}'
-                    WITH (format = 'csv', empty_string_as_null = true)
-                    RETURN SUMMARY;
-                """,
-        ).execute({})
-
+from airflow.operators.bash import BashOperator
+from airflow.models import Variable
+
+
+@dag(dag_id="nyc-taxi-parquet",
+schedule_interval='@monthly',
+start_date=datetime(2009, 1, 1),
+catchup=True)
+def taskflow():
+    @task(task_id='get_month_year')
+    def get_month_year(ds=None):
+        currentMonth = int(ds.split('-')[1]) - 2
+        currentYear = int(ds.split('-')[0])
+        if currentMonth == 12:
+            currentYear = currentYear - 1
+        if currentMonth < 10:
+            currentMonth = f'0{currentMonth}'
+
+        return f'_{currentYear}-{currentMonth}'
+
+
+    @task(task_id='download_file_s3')
+    def download_from_s3_csv(file_date):
+        destination_path = Variable.get("destination_path")
+        s3_path = Variable.get("s3_path")
+        file_name = f'yellow_tripdata{file_date}'
+        file_path = f'{destination_path}{file_name}.csv'
+        BashOperator(
+            task_id='download_from_s3_csv',
+            bash_command=f'parquet-tools csv "{s3_path}{file_name}.parquet" > "{file_path}"',
+            ).execute({})
+        return file_path
+
+    @task(task_id='process_new_files')
+    def process_new_files(new_csv_file):
         PostgresOperator(
-            task_id=f"log_{file_name}",
-            postgres_conn_id="cratedb_demo_connection",
-            sql=Path("include/taxi-insert.sql").read_text(encoding="utf-8"),
-        ).execute({})
+                task_id="copy_new_csv_file",
+                postgres_conn_id="cratedb_demo_connection",
+                sql=f"""
+                        COPY nyc_taxi.load_trips_staging
+                        FROM '{new_csv_file}'
```

Comment:
Is `new_csv_file` a local file or an S3 URL? Will this work if CrateDB and Airflow aren't running on the same file system?

Reply:
I haven't considered that. I was assuming this was checking the local file in Airflow, which was working for me. But I'll have a further look.